05_stringer/marksomething/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStringer(t *testing.T) {
+	testCases := map[string]struct {
+		arg      IPAddr
+		expected string
+	}{
+		"Empty": {
+			arg:      IPAddr{},
+			expected: "0.0.0.0",
+		},
+		"Partial": {
+			arg:      IPAddr{127},
+			expected: "127.0.0.0",
+		},
+		"LocalHost": {
+			arg:      IPAddr{127, 0, 0, 1},
+			expected: "127.0.0.1",
+		},
+		"Common Local": {
+			arg:      IPAddr{192, 168, 0, 1},
+			expected: "192.168.0.1",
+		},
+		"Common Local Broadcast": {
+			arg:      IPAddr{192, 168, 0, 255},
+			expected: "192.168.0.255",
+		},
+	}
+
+	for testName, tC := range testCases {
+		testCase := tC
+		t.Run(testName, func(t *testing.T) {
+			actual := fmt.Sprint(testCase.arg)
```

Comment:
You just implemented the method String(), so best to call String() in test cases . e.g. 'expected := testCase.arg.String()

Reply:
I used fmt.* here as the title of the lab was stringer so i figured it made sense to call a function that used the stringer interface.
I've hedged by bets now testing both.